Core/Src/utils.c

```diff
@@ -141,12 +130,19 @@ bool calc_validate_upshift(gear_t current_gear, U8 fast_clutch, U8 slow_clutch)
 		case GEAR_2:
 		case GEAR_3:
 		case GEAR_4:
-			tcm_data.target_gear = current_gear + 1;
+			tcm_data.target_gear = current_gear + 2;
 			tcm_data.target_RPM = calc_target_RPM(tcm_data.target_gear);
```

Comment:
There is no longer any reason to calculate the target RPM because we are always going to return true as long as we are not shifting into first from neutral